### PR TITLE
Update to the Bot API 6.4 version

### PR DESCRIPTION
**Pull request by tupchienko**
- Added the field platform, the optional parameter options to the method openLink and the methods showScanQrPopup, closeScanQrPopup, readTextFromClipboard to the class [WebApp](https://core.telegram.org/bots/webapps#initializing-web-apps).
- Added the events qrTextReceived, clipboardTextReceived.
- Added parameters in the invoiceClosed event.

### CODE REVIEW

File: types/telegram-web-app/index.d.ts [hunk 1]
```diff
@@ -32,6 +32,10 @@ interface WebApp {
      * The color scheme currently used in the Telegram app. Either “light” or
      * “dark”. Also available as the CSS variable var(--tg-color-scheme).
      */
+    platform: string;
```
Comment (MKRhere) [inline types/telegram-web-app/index.d.ts:35]: JSDoc should go above the property it describes; this edit has made the JSDoc for color move out of place.

Comment (tup4ienko) [reply types/telegram-web-app/index.d.ts:35]: I will fix it

Comment (tup4ienko) [reply types/telegram-web-app/index.d.ts:34]: Fixed @MKRhere

File: types/telegram-web-app/index.d.ts [hunk 2]
```diff
@@ -157,12 +169,15 @@ interface WebApp {
      */
     sendData(data: string): void;
     /**
-     * A method that opens a link in an external browser. The Web App will not
-     * be closed. Note that this method can be called only in response to the
-     * user interaction with the Web App interface (e.g. click inside the Web
-     * App or on the main button)
+     * A method that opens a link in an external browser.
```
Comment (KnorpelSenf) [inline types/telegram-web-app/index.d.ts:172]: What's the point of this diff? It looks like it just reformats the comment. In that case, please revert it.

File: types/telegram-web-app/index.d.ts [hunk 3]
```diff
@@ -157,12 +169,15 @@ interface WebApp {
...
+     * The Web App will not be closed.
+     * If the optional options parameter is passed with the field
+     * @param try_instant_view the link will be opened in Instant View mode if possible.
+     *
+     * Note that this method can be called only in response to user interaction with
+     * the Web App interface (e.g. a click inside the Web App or on the main button)
      */
-    openLink(url: string): void;
+    openLink(url: string, try_instant_view?: boolean): void;
```
Comment (KnorpelSenf) [inline types/telegram-web-app/index.d.ts:180]: This should be
```suggestion
    openLink(url: string, options?: { try_instant_view?: boolean }): void;
```

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]
Thanks for your work! I'm surprised to see we were missing changes from the Bot API 6.1 update. Glad that you fixed this.

I just have a small correction:

**Review by KnorpelSenf — APPROVED** [review]
LGTM!

**Comment by KnorpelSenf** [comment]
Needs maintainer review? Why is that?

**Review by KnorpelSenf — APPROVED** [review]

**Comment by KnorpelSenf** [comment]
@MKRhere perhaps it's waiting for you?

**Review by MKRhere — CHANGES_REQUESTED** [review]
Sorry, one last minor change; could you change the first line to latest version number?

```TS
// Type definitions for non-npm package telegram-web-app 6.4
```

**Comment by tup4ienko** [comment]
@KnorpelSenf @MKRhere I fixed the version in the documentation.

**Review by tupchienko — COMMENTED** [review]
I fixed this.

**Review by sheetalkamat — APPROVED** [review]

**Comment by MKRhere** [comment]
Ready to merge

**Comment by MKRhere** [comment]
Thanks @tup4ienko!

**Pull request merged**